src/orion/core/cli/checks/presence.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.core.cli.checks.presence` -- Presence stage for database checks
+===========================================================================
+
+.. module:: presence
+    :platform: Unix
+    :synopsis: Checks for the creation of a `Database` object.
+
+"""
+
+from orion.core.utils.decorators import register_check
+from orion.core.utils.exceptions import CheckError
+
+
+class _Checks:
+    checks = []
+
+
+class PresenceStage:
+    """The presence stage of the checks."""
+
+    def __init__(self, experiment_builder, cmdargs):
+        """Create an instance of the stage.
+
+        Parameters
+        ----------
+        experiment_builder: `ExperimentBuilder`
+            An instance of `ExperimentBuilder` to fetch configs.
+
+        """
+        self.builder = experiment_builder
+        self.cmdargs = cmdargs
+        self.db_config = None
+
+    @staticmethod
+    def checks():
+        """Return the registered checks."""
+        return _Checks.checks
+
+    @register_check(_Checks.checks, "check if oríon's default options are present... ")
+    def check_default_config(self):
+        """Check if database supports delete operation."""
+        config = self.builder.fetch_default_options()
+
+        if 'database' not in config:
+            return "Skipping", "No default configuration found for database."
+
+        self.db_config = config['database']
+
+        return "Success", ""
+
+    @register_check(_Checks.checks, "check if oríon's environment variables have been set... ")
+    def check_environment_vars(self):
+        """Check if database supports delete operation."""
+        config = self.builder.fetch_env_vars()
+
+        config = config['database']
+        names = ['type', 'name', 'host']
+
+        for name in names:
+            if name not in config or config[name] is None:
+                return "Skipping", "Missing {} environment variable.".format(name)
```

Comment:
Not all of them may be set. The set should support partial definition using env vars.